### ISSUE TITLE
gcd documentation mixes up x and y WRT value of a

### ISSUE THREAD

**Issue by CharnelMouse** [factor/factor#2924]
The documentation for gcd ( x y -- a d ) states the following:
> Computes the positive greatest common divisor d of x and y, and another value a satisfying: a*y = d mod x

This is incorrect: a is the required multiplier for a*x = d mod y.
Example code to show this:
```factor
USE: backtrack
: check-gcd ( x y -- ? ) 2dup gcd [ * ] dip - swap mod 0 = ;
: check-gcd-opp-spec ( x y -- ? ) [ swap ] [ gcd ] 2bi [ * ] dip - swap mod 0 = ;
[ 1 30 [a..b] amb 1 30 [a..b] amb 2dup check-gcd not must-be-true 2array . t ] [ "All good." print ] if-amb drop
[ 1 30 [a..b] amb 1 30 [a..b] amb 2dup check-gcd-opp-spec not must-be-true 2array . t ] [ "All good." print ] if-amb drop
```

**Comment by mrjbq7** [comment]
This prints

```
{ 2 1 }
All good.
```

Does that mean the only inputs that fails for are ``2 1 gcd`` ?

**Comment by CharnelMouse** [comment]
It's just the first/simplest failing case that backtrack finds. If you exclude cases where an input is equal to 1, it finds { 2 5 } to be a failing case.

**Comment by mrjbq7** [comment]
Fixed in 859077b670ea6f9d4ed34382b566947480700b6f.
